streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -293,17 +288,12 @@ public void cleanup() throws IOException {
             task = null;
         }
         Utils.delete(BASE_DIR);
-        mockito.finishMocking();
     }
 
     @Test
-    public void shouldThrowLockExceptionIfFailedToLockStateDirectory() throws IOException {
-        stateDirectory = EasyMock.createNiceMock(StateDirectory.class);
-        EasyMock.expect(stateDirectory.lock(taskId)).andReturn(false);
-        EasyMock.expect(stateManager.changelogPartitions()).andReturn(Collections.emptySet());
-        stateManager.registerStore(stateStore, stateStore.stateRestoreCallback, null);
-        EasyMock.expectLastCall();
-        EasyMock.replay(stateDirectory, stateManager);
+    public void shouldThrowLockExceptionIfFailedToLockStateDirectory() {
+        stateDirectory = mock(StateDirectory.class);
```

Comment:
So, I tried with explicitly closing the "old" stateDirectory, but that was complaining because the lockStateDirectory method was never called on it:
```
    @Override
    public void close() {
        if (hasPersistentStores) {
            try {
                stateDirLock.release(); <--- THIS IS NULL;
                stateDirLockChannel.close();

                stateDirLock = null;
                stateDirLockChannel = null;
            } catch (final IOException e) {
               ...
            }
        }
    }
```
I am not too familiar with the StateDirectory class itself, what underlying resources could be leaking and I ought to pay attention to closing?

Reply:
In general, an implementation of `AutoClosable` should always be closed. If closing does not work, let's remove the creation of the state directory from `setup()`.

Reply:
Okay, let me try this out

Reply:
Heya @cadonna! I had another pass. The problem here is that the tests which appear to be properly using the state directory don't actually lock it. This means that when I try to close it I get the error above. As such, I decided to explicitly lock the state directory during setup (added the call + explanation to the setup method). Let me know if you disagree with this approach!